src/DI/Extension.php

```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types = 1);
+
+namespace Pd\SecurityHeaders\DI;
+
+final class Extension extends \Nette\DI\CompilerExtension
+{
+
+	/**
+	 * @var string
+	 */
+	private $presenterHook = HeadersSetup::class;
+
+
+	public function loadConfiguration()
+	{
+		parent::loadConfiguration();
+
+		$config = $this->getConfig();
+
+		if (isset($config['presenterHook'])) {
+			$this->presenterHook = $config['presenterHook'];
+		}
+
+		$presenterHookReflection = new \ReflectionClass($this->presenterHook);
+
+		if ( ! $presenterHookReflection->implementsInterface(IOnPresenterListener::class)) {
```

Comment:
Viděl jsem, že jsi tady před tím, místo té reflexe, měl `$this->presenterHook instanceof IOnPresenterListener`. Proč to tak nefunguje?

```php
if ( ! $this->presenterHook instanceof IOnPresenterListener) {
    throw...;
}
```
toto by mělo fungovat ne? Je to lepší než přes reflexi.

Reply:
On tam totiž je název classy jako string, její instance je až když se to vytahuje z kontejneru. PHPStan mě na to upozornil.